src/components/DragAndDropFile.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { Component, ChangeEvent, DragEvent } from 'react';
+
+type DragAndDropFileProps = {
+    onFileInputChange: (file: File | null) => void;
+};
+
+type DragAndDropFileState = {
+    dragging: boolean;
+};
+
+class DragAndDropFile extends Component<DragAndDropFileProps, DragAndDropFileState> {
+    state: DragAndDropFileState = {
+        dragging: false,
+    };
+
+    dropRef: any;
+    fileInput: any;
+    dragCounter: number;
+
+    constructor(props: DragAndDropFileProps) {
+        super(props);
+        this.dropRef = React.createRef();
+        this.dragCounter = 0;
+
+        this.handleFileDrag = this.handleFileDrag.bind(this);
+        this.handleFileDragIn = this.handleFileDragIn.bind(this);
+        this.handleFileDragOut = this.handleFileDragOut.bind(this);
+        this.handleFileDrop = this.handleFileDrop.bind(this);
+    }
+
+    componentDidMount(): void {
+        this.dropRef.current.addEventListener('dragenter', this.handleFileDragIn);
+        this.dropRef.current.addEventListener('dragleave', this.handleFileDragOut);
+        this.dropRef.current.addEventListener('dragover', this.handleFileDrag);
+        this.dropRef.current.addEventListener('drop', this.handleFileDrop);
+    }
+
+    componentWillUnmount(): void {
+        this.dropRef.current.removeEventListener('dragenter', this.handleFileDragIn);
+        this.dropRef.current.removeEventListener('dragleave', this.handleFileDragOut);
+        this.dropRef.current.removeEventListener('dragover', this.handleFileDrag);
+        this.dropRef.current.removeEventListener('drop', this.handleFileDrop);
+    }
+
+    handleFileInputChange(event: ChangeEvent<HTMLInputElement>): void {
+        const file = event.target.files ? event.target.files[0] : null;
+        this.props.onFileInputChange(file);
+        console.log('handle file input change');
+    }
+
+    handleFileDragIn(event: DragEvent<HTMLDivElement>): void {
+        event.preventDefault();
+        event.stopPropagation();
+        this.dragCounter++;
+        if (event.dataTransfer.items && event.dataTransfer.items.length > 0) {
+            this.setState({ dragging: true });
+        }
+    }
+
+    handleFileDragOut(event: DragEvent<HTMLDivElement>): void {
+        event.preventDefault();
+        event.stopPropagation();
+        this.dragCounter--;
+        if (this.dragCounter <= 0) {
+            this.setState({ dragging: false });
+        }
+    }
+
+    handleFileDrag(event: DragEvent<HTMLDivElement>): void {
+        event.preventDefault();
+        event.stopPropagation();
+    }
+
+    handleFileDrop(event: DragEvent<HTMLDivElement>): void {
```

Comment:
I know I wrote this but this would only allow the user to drag only one file in right? Should we change to handle multiple file drops or should we update that once we have that functionality implemented in the `FileTransfer.tsx` component?

Reply:
I think we should change it later and stick with one for now. But when we do change it later, the user isn't able to drop 2 files at the same time, and `handleFileDrop` should only handle the event of the drop, so I don't think this function will have to worry about multiple files, but we will probably have to edit the `files[0]` logic.